lib/tapioca/gem/listeners/foreign_constants.rb

```diff
@@ -35,10 +35,8 @@ def on_scope(event)
               # base constant. Then, generate RBIs as if the base constant is extending the mixin,
               # which is functionally equivalent to including or prepending to the singleton class.
               if !name && constant.singleton_class?
-                name = constant_name_from_singleton_class(constant)
-                next unless name
-
-                constant = T.cast(constantize(name), Module)
+                attached_class = attached_class_of(constant)
+                constant = attached_class if attached_class
```

Comment:
Previously we were skipping the rest of the processing if the attached class wasn't found (if the `name` wasn't available). In the new version we continue processing if `attached_class` is nil, which might lead to changes in behaviour in the future if more code is added after this block.

Reply:
Oh good catch. Thanks for pointing that out.